File: Marcs_cakewalk.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    static long marcsCakewalk(int[] calorie) {
        // Complete this function
        //Integer[] cubes = new Integer[] {calorie};
        //Arrays.sort(cubes, Collections.reverseOrder());


        Arrays.sort(calorie);
        long sum =0;
        int j=0;
        for(int i=calorie.length-1;i>=0;i--)
        {
            sum = sum + (long) (calorie[i] * Math.pow(2,j));
            j++;
            //System.out.println("Value="+calorie[i] * Math.pow(2,i));
        }
        return sum;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] calorie = new int[n];
        for(int calorie_i =0; calorie_i < n; calorie_i++){
            calorie[calorie_i] = in.nextInt();
        }
        long result = marcsCakewalk(calorie);
        System.out.println(result);
        in.close();
    }
}

File: Marcs_cakewalk.py
#!/bin/python3

import sys

def marcsCakewalk(calorie):
    # Complete this function
    calorie.sort()
    i=len(calorie)-1
    sum = 0
    j =0
    while i>=0:
        sum += calorie[i] * (2**j)
        #print(i)
        #print(j)
        #print(calorie[i] * (2^j))
        j = j+1
        i = i-1
    return sum
        

if __name__ == "__main__":
    n = int(input().strip())
    calorie = list(map(int, input().strip().split(' ')))
    #print(calorie)
    result = marcsCakewalk(calorie)
    print(result)


